### PR TITLE
Add possibility to define performance data as regular expression

### PR DESCRIPTION
**Pull request by dbenesj**
As it is possible to define elements as regular expression, there was no possibility to get such value also as performance data.

So in this PR, I have added such functionality.
Implementation was inspired by the element regex and follows exactly the same logic.

### CODE REVIEW

**Comment by phrawzty** [comment]
Hi there! Thanks for the PR. To be honest, it's been a long time since I touched either this code or Nagios in general. I am therefore not in a good position to say whether this functionality is a good idea or not. :)

That said, I must ask, why are you are you renaming `:perf` to `:perf_string`? Also, in general, if this is to be renamed, why isn't it renamed everywhere (see [here](https://github.com/phrawzty/check_http_json/pull/50/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R32) and [here](https://github.com/phrawzty/check_http_json/pull/50/files#diff-52da28f441b8c346d999d2f496d31cd1aeda6bfc760b16853b338b2f304b846eR44)).

**Comment by phrawzty** [comment]
Hi there! Thanks for the PR. To be honest, it's been a long time since I touched either this code or Nagios in general. I am therefore not in a good position to say whether this functionality is a good idea or not. :)

That said, I must ask, why are you are you renaming `:perf` to `:perf_string`? Also, in general, if this is to be renamed, why isn't it renamed everywhere (see [here](https://github.com/phrawzty/check_http_json/pull/50/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R32) and [here](https://github.com/phrawzty/check_http_json/pull/50/files#diff-52da28f441b8c346d999d2f496d31cd1aeda6bfc760b16853b338b2f304b846eR44)).

**Comment by dbenesj** [comment]
Thanks for taking a look even when you are not using it lately.

Reason for adding this functionality is, that before this PR, we were not able to graph value that is being checked by the `--element_regex` parameter.

The rename of internal variable from `:perf` to `:perf_string`, but preserving command line parameter `--perf` was just because it is done exactly the same already for `--element` [here](https://github.com/phrawzty/check_http_json/blob/main/check_http_json.rb#L314-L322).
So I was trying to be more in line with the previous implementation. Initialy I did not done the rename, but when comparing with the `--element` and `--element_regex` I found out that having just the variable `:perf` you would not be sure whether it is a string or the regular expression.
If you prefer to keep the variable name `:perf`, I'm totally fine with that and will revert that part :wink:

File: check_http_json.rb [hunk 1]
```diff
@@ -438,6 +448,10 @@ def sanity_check(options)
         error_msg.push('Must specify either a result string OR result regular expression.')
     end
 
+    if options[:perf_string] and options[:perf_regex] then
+        error_msg.push('Must specify either an perf string OR an perf regular expression.')
```
Comment (phrawzty) [inline check_http_json.rb:452]:
```suggestion
        error_msg.push('Must specify either a perf string OR a perf regular expression.')
```

**Review by phrawzty — CHANGES_REQUESTED** [review]

**Review by phrawzty — APPROVED** [review]

**Comment by dbenesj** [comment]
Thank you very much as well!

**Pull request merged**